generated/nixnet/nixnet_service.cpp

```diff
@@ -166,6 +167,78 @@ namespace nixnet_grpc {
     }
   }
 
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiXnetService::ConvertFramesToSignalsSinglePoint(::grpc::ServerContext* context, const ConvertFramesToSignalsSinglePointRequest* request, ConvertFramesToSignalsSinglePointResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto session_ref_grpc_session = request->session_ref();
+      nxSessionRef_t session_ref = session_repository_->access_session(session_ref_grpc_session.id(), session_ref_grpc_session.name());
+      auto frame_buffer = convert_from_grpc<u8>(request->frame_buffer());
+      auto number_of_bytes_for_frames = frame_buffer.size();
+      u32 size_of_value_buffer = request->size_of_value_buffer();
+      u32 size_of_timestamp_buffer = request->size_of_timestamp_buffer();
+      response->mutable_value_buffer()->Resize(size_of_value_buffer, 0);
+      f64* value_buffer = response->mutable_value_buffer()->mutable_data();
+      response->mutable_timestamp_buffer()->Resize(size_of_timestamp_buffer, 0);
+      nxTimestamp100ns_t* timestamp_buffer = reinterpret_cast<nxTimestamp100ns_t*>(response->mutable_timestamp_buffer()->mutable_data());
+      auto status = library_->ConvertFramesToSignalsSinglePoint(session_ref, frame_buffer, number_of_bytes_for_frames, value_buffer, size_of_value_buffer, timestamp_buffer, size_of_timestamp_buffer);
+      response->set_status(status);
+      if (status_ok(status)) {
+      }
+      return ::grpc::Status::OK;
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiXnetService::ConvertSignalsToFramesSinglePoint(::grpc::ServerContext* context, const ConvertSignalsToFramesSinglePointRequest* request, ConvertSignalsToFramesSinglePointResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto session_ref_grpc_session = request->session_ref();
+      nxSessionRef_t session_ref = session_repository_->access_session(session_ref_grpc_session.id(), session_ref_grpc_session.name());
+      auto value_buffer = const_cast<f64*>(request->value_buffer().data());
+      u32 size_of_value_buffer = static_cast<u32>(request->value_buffer().size());
```

Comment:
note: This calculation is incorrect. XNET expects number of bytes in `size_of_value_buffer` parameter, but we are passing in number of f64 elements here.